Examples/Algorithms/Utilities/src/PrototracksToTracks.cpp

```diff
@@ -0,0 +1,91 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Utilities/ProtoTracksToTracks.hpp"
+
+#include "Acts/Surfaces/PerigeeSurface.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/MeasurementCalibration.hpp"
+#include "ActsExamples/EventData/ProtoTrack.hpp"
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+#include "ActsExamples/Utilities/EventDataTransforms.hpp"
+
+#include <algorithm>
+
+namespace ActsExamples {
+
+PrototracksToTracks::PrototracksToTracks(Config cfg, Acts::Logging::Level lvl)
+    : IAlgorithm("PrototracksToTracks", lvl), m_cfg(std::move(cfg)) {
+  m_outputTracks.initialize(m_cfg.outputTracks);
+  m_inputProtoTracks.initialize(m_cfg.inputProtoTracks);
+  m_inputMeasurements.initialize(m_cfg.inputMeasurements);
+}
+
+ProcessCode PrototracksToTracks::execute(const AlgorithmContext& ctx) const {
+  auto trackContainer = std::make_shared<Acts::VectorTrackContainer>();
+  auto mtj = std::make_shared<Acts::VectorMultiTrajectory>();
+  TrackContainer tracks(trackContainer, mtj);
+
+  PassThroughCalibrator calibratorImpl;
+  MeasurementCalibratorAdapter calibrator(calibratorImpl,
+                                          m_inputMeasurements(ctx));
+
+  auto refSurface = Acts::Surface::makeShared<Acts::PerigeeSurface>(
+      Acts::Vector3{0., 0., 0.});
+  const auto& prototracks = m_inputProtoTracks(ctx);
+  ACTS_DEBUG("Received " << prototracks.size() << " prototracks");
+
+  float avgSize = 0;
+  std::size_t minSize = std::numeric_limits<std::size_t>::max();
+  std::size_t maxSize = 0;
+
+  for (const auto& protoTrack : prototracks) {
+    if (protoTrack.empty()) {
+      continue;
+    }
+
+    avgSize += protoTrack.size();
+    minSize = std::min(minSize, protoTrack.size());
+    maxSize = std::max(maxSize, protoTrack.size());
+
+    auto track = tracks.makeTrack();
+    for (auto idx : protoTrack) {
+      auto trackStateProxy =
+          track.appendTrackState(Acts::TrackStatePropMask::Calibrated);
+
+      IndexSourceLink sl(Acts::GeometryIdentifier{}, idx);
+
+      calibrator.calibrate(ctx.geoContext, ctx.calibContext,
+                           Acts::SourceLink{sl}, trackStateProxy);
+      trackStateProxy.typeFlags().set(Acts::TrackStateFlag::MeasurementFlag);
+    }
+
+    track.nMeasurements() = protoTrack.size();
+    track.nHoles() = 0;
+    track.nOutliers() = 0;
+    track.setReferenceSurface(refSurface->getSharedPtr());
```

Comment:
I think the ref surface should be omitted

Reply:
A reference surface is required by the `CKFPerformanceWriter`

Reply:
But in that case we should change the `CKFPerformanceWriter` IMO. Setting information we don't have can be a footgun